api/node/server/server.go

```diff
@@ -0,0 +1,198 @@
+	}
+	id := types.BytesToATXID(idBytes)
+	atx, err := s.atxService.Atx(ctx, id)
+	switch {
+	case errors.Is(err, activation.ErrNotFound):
+		return GetActivationAtxAtxId404Response{}, nil
+	case err != nil:
+		return nil, err
+	}
+
+	return GetActivationAtxAtxId200JSONResponse{
+		ID:           request.AtxId,
+		NumUnits:     atx.NumUnits,
+		PublishEpoch: atx.PublishEpoch.Uint32(),
+		Sequence:     &atx.Sequence,
+		SmesherID:    hex.EncodeToString(atx.SmesherID.Bytes()),
+		TickCount:    atx.TickCount,
+		Weight:       atx.Weight,
+	}, nil
+}
+
+// GetActivationLastAtxNodeId implements StrictServerInterface.
+func (s *Server) GetActivationLastAtxNodeId(
+	ctx context.Context,
+	request GetActivationLastAtxNodeIdRequestObject,
+) (GetActivationLastAtxNodeIdResponseObject, error) {
+	id, err := models.ParseNodeID(request.NodeId)
+	if err != nil {
+		msg := err.Error()
+		return GetActivationLastAtxNodeId400PlaintextResponse{
+			Body:          bytes.NewBuffer([]byte(msg)),
+			ContentLength: int64(len(msg)),
+		}, nil
+	}
+
+	atxid, err := s.atxService.LastATX(ctx, id)
+	switch {
+	case errors.Is(err, activation.ErrNotFound):
+		return GetActivationLastAtxNodeId404Response{}, nil
+	case err != nil:
+		return nil, err
+	}
+
+	return GetActivationLastAtxNodeId200JSONResponse{
+		ID:           hex.EncodeToString(atxid.ID().Bytes()),
+		NumUnits:     atxid.NumUnits,
+		PublishEpoch: atxid.PublishEpoch.Uint32(),
+		Sequence:     &atxid.Sequence,
+		SmesherID:    hex.EncodeToString(atxid.SmesherID.Bytes()),
+		TickCount:    atxid.TickCount,
+		Weight:       atxid.Weight,
+	}, nil
+}
+
+// GetActivationPositioningAtxEpoch implements StrictServerInterface.
+func (s *Server) GetActivationPositioningAtxPublishEpoch(
+	ctx context.Context,
+	request GetActivationPositioningAtxPublishEpochRequestObject,
+) (GetActivationPositioningAtxPublishEpochResponseObject, error) {
+	id, err := s.atxService.PositioningATX(ctx, types.EpochID(request.PublishEpoch))
+	if err != nil {
+		return nil, err
+	}
+
+	return GetActivationPositioningAtxPublishEpoch200JSONResponse{
+		ID: hex.EncodeToString(id.Bytes()),
+	}, nil
+}
+
+// PostPublishProtocol implements StrictServerInterface.
+func (s *Server) PostPublishProtocol(
+	ctx context.Context,
+	request PostPublishProtocolRequestObject,
+) (PostPublishProtocolResponseObject, error) {
+	blob, err := io.ReadAll(request.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	s.publisher.Publish(ctx, string(request.Protocol), blob)
+	return PostPublishProtocol200Response{}, nil
+}
+
+// PostPoet implements StrictServerInterface.
+func (s *Server) PostPoet(ctx context.Context, request PostPoetRequestObject) (PostPoetResponseObject, error) {
+	var proof types.PoetProofMessage
+	blob, err := io.ReadAll(request.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := codec.Decode(blob, &proof); err != nil {
+		msg := err.Error()
+		return PostPoet400PlaintextResponse{
+			Body:          bytes.NewBuffer([]byte(msg)),
```

Comment:
i honestly am not a big fan of returning plaintext error messages. just returning a status code should be enough imo.

Reply:
IMHO, knowing the error on the client side when something doesn't work might be helpful for debugging.